p8e-api/src/main/kotlin/io/provenance/engine/grpc/observers/EnvelopeEventObserver.kt

```diff
@@ -136,7 +138,31 @@ class EnvelopeEventObserver(
                 }
                 else -> queuer.error(IllegalStateException("Unknown action received by server ${value.action.name}").statusRuntimeException())
             }
+        } else {
+            // send a dummy event back on connect to force headers through in order to satisfy the ConnectionTimeoutInterceptor
+            if (clientSupportsEnvelopeWatchConnected()) {
+                EnvelopeEvent.newBuilder()
+                    .setEvent(EventType.ENVELOPE_WATCH_CONNECTED)
+                    .setAction(CONNECT)
+                    .setPublicKey(
+                        PK.SigningAndEncryptionPublicKeys.newBuilder()
+                            .setSigningPublicKey(queuerKey!!.publicKey.toPublicKeyProto())
+                            .setEncryptionPublicKey(queuerKey!!.publicKey.toPublicKeyProto())
+                    ).build()
+                    .let(queuer::queue)
+            } else {
+                logger().info("Skipping ENVELOPE_WATCH_CONNECTED event for unsupported client (version ${clientVersion()})")
+            }
+        }
+    }
+
+    private fun clientSupportsEnvelopeWatchConnected(): Boolean = try {
+        SemVer.parse(clientVersion()).let {
+            it.compareTo(SemVer.parse("0.8.22")) > 0
```

Comment:
Maybe put this in a const with a comment to the commit where it was added?

Reply:
Good call, added that